web-server/src/content/DoraMetrics/AIAnalysis/AIAnalysis.tsx

```diff
@@ -0,0 +1,306 @@
+import {
+  AutoAwesomeRounded,
+  CheckRounded,
+  ChevronLeftRounded,
+  ChevronRightRounded
+} from '@mui/icons-material';
+import { TabContext, TabList, TabPanel } from '@mui/lab';
+import {
+  Alert,
+  Button,
+  Card,
+  CircularProgress,
+  IconButton,
+  InputAdornment,
+  MenuItem,
+  Select,
+  Tab,
+  TextField,
+  useTheme
+} from '@mui/material';
+import { AxiosError } from 'axios';
+import copy from 'copy-to-clipboard';
+import { enqueueSnackbar } from 'notistack';
+import { useEffect, useMemo } from 'react';
+import { AiOutlineOpenAI } from 'react-icons/ai';
+import { FaMeta } from 'react-icons/fa6';
+import ReactMarkdown from 'react-markdown';
+import gfm from 'remark-gfm';
+import v from 'voca';
+
+import { FlexBox } from '@/components/FlexBox';
+import { Line } from '@/components/Text';
+import { useAxios } from '@/hooks/useAxios';
+import { useEasyState } from '@/hooks/useEasyState';
+import { usePrevious } from '@/hooks/usePrevious';
+import { DoraAiAPIResponse, doraMetricsSlice } from '@/slices/dora_metrics';
+import { useDispatch, useSelector } from '@/store';
+import { depFn, noOp } from '@/utils/fn';
+
+enum Model {
+  GPT4o = 'GPT4o',
+  LLAMA3p1450B = 'LLAMA3p1450B',
+  LLAMA3p1405B = 'LLAMA3p1405B',
+  LLAMA3p170B = 'LLAMA3p170B'
+}
+
+const openAiModels = new Set([Model.GPT4o]);
+
+const modelLabelsMap: Record<Model, string> = {
+  GPT4o: 'GPT 4o',
+  LLAMA3p1450B: 'Llama 3.1 / 405B',
+  LLAMA3p1405B: 'Llama 3.1 / 405B',
+  LLAMA3p170B: 'Llama 3.1 / 70B'
+};
+
+export const AIAnalysis = () => {
+  const [models, { loading: loadingModels, fetch: fetchModels }] = useAxios<
+    Model[]
+  >(`/internal/ai/models`, { manual: true });
+
+  const dispatch = useDispatch();
+  const theme = useTheme();
+  const doraData = useSelector((s) => s.doraMetrics.metrics_summary);
+  const savedSummary = useSelector((s) => s.doraMetrics.ai_summary);
+  const savedToken = useSelector((s) => s.doraMetrics.ai_summary_token);
+
+  const [response, { loading: loadingData, fetch: loadData }] =
+    useAxios<DoraAiAPIResponse>(`/internal/ai/dora_metrics`, {
+      manual: true,
+      onError: noOp
+    });
+
+  const data = useMemo(
+    () => response || savedSummary,
+    [response, savedSummary]
+  );
+
+  const selectedModel = useEasyState<Model>(Model.GPT4o);
+  const token = useEasyState<string>('');
+  const selectedTab = useEasyState<string>('0');
+
+  useEffect(() => {
+    fetchModels();
+  }, [fetchModels]);
+
+  const prevSavedToken = usePrevious(savedToken);
+  useEffect(() => {
+    if (token.value || prevSavedToken === savedToken) return;
+    depFn(token.set, savedToken);
+  }, [prevSavedToken, savedToken, token.set, token.value]);
+
+  return (
+    <FlexBox
+      col
+      gap1
+      sx={{ ['[data-lastpass-icon-root]']: { display: 'none' } }}
```

Comment:
what's this?

Reply:
It's to remove the password manager from showing its autofill icon in the input field.
It's... unfortunately the recommended way to deal with it.

Reply:
damn, this is interesting !